### PR TITLE
Prop editor input improvements

### PR DESCRIPTION
**Pull request by toger5**
there are actually three things in this pr. (I can separate them if desired)

### Shortcuts for spin_slider
the spin slider had an exp(), speed dependant, behaviour (which was hard to control)
now it is easy to control it's behaviour by shortcuts (ctrl = int steps, shift = factor 0.1)
ctrl_shift_showcase

Also the step per int is now always consistent (6 pixels per int) and slow enough to actually make small steps.
also pressing control during spinning rounds the value to the next int immediately.

There is an issue on OS X, that control click is changed to right click... This prohibits you from holding control and than start to drag (with int steps), instead you first have to drag, and than (when the mouse button is hold down) press ctrl. It works fine but if someone has a better solution would be great.
### Better tab control
this was hard to implement there might be an easier solution (took me a long time to figure out what signals I should use) so any help/feedback is desired.

godot_showcase_tab_prop_editor

 - tab now also enters the text and you go to the next input AND activated the value_input (the modal popup with the textfield) (as seen in the gif, before you had to: enter, down arrow, enter, text input, enter....)
 - enter closes the modal value_input
 - esc closes the modal value_input
 - click outside the input, closes the modal value_input

### fixed showing grabber
The grabber was still shown when editing the number with the value_input.
which looked buggy

### Removed the grey highlighting on the left of the inspector
Since the one on the items themselves is more than enough Imo. (looks better I think, it's just an aesthetic thing)
Can be seen in the gifs too.
@djrm you told me I should not add sneaky changes (which you are right!) I made it a separate commit. so we can easily remove it.

### CODE REVIEW

**Comment by djrm** [comment]
@toger5 im currently working on the Inspector visual design, and i already removed the section highlighting, so..... i don't think it is necessary.

**Comment by toger5** [comment]
@djrm than it would even lead to merge conflicts. ;) 
so I remove that commit. Thank you!!

File: editor/editor_inspector.cpp [hunk 1]
```diff
@@ -1021,7 +1021,7 @@ void EditorInspectorSection::setup(const String &p_section, const String &p_labe
 			vbox->hide();
 		}
 	}
-		//	void editor_set_section_unfold(const String &p_section, bool p_unfolded);
+	//	void editor_set_section_unfold(const String &p_section, bool p_unfolded);
```
Comment (toger5) [inline editor/editor_inspector.cpp]: srsly.. this is literally what my local clang made me do ... now travis hates it

**Comment by toger5** [comment]
@djrm super curious about how you handle the sub-resource editor. Will it become easier to see?

**Comment by djrm** [comment]
@toger5 my current proposal is something like this (colors are not final), it makes subresource section very easy to see, and reduces the margin needed, so you can easily have up to 10 (maybe less) i would say depth levels without having a tiny sub inspector.

peek 2018-05-20 20-26

**Comment by TheYokai** [comment]
Tab support is very nice!

**Comment by toger5** [comment]
can this get merged before there are any conflicts?

**Comment by toger5** [comment]
@akien-mga rebased

**Pull request merged**